pkg/evpn/evpn.go

```diff
@@ -131,9 +131,19 @@ func UnmarshalEVPNNLRI(b []byte) (*Route, error) {
 				return nil, err
 			}
 		case 5:
-			n.RouteTypeSpec, err = UnmarshalEVPNIPPrefix(b[p : p+l])
-			if err != nil {
-				return nil, err
+			switch l {
+				case 34:
+					n.RouteTypeSpec, err = UnmarshalEVPNIPv4Prefix(b[p : p+l])
```

Comment:
These 2 Unmarshal func looks like do the same thing, please use  common  func for both and pass the prefix length as a parameter.